### PR TITLE
Testing ordu hooks

### PR DESCRIPTION
**Pull request by vitorclelis96**

### CODE REVIEW

File: src/newrelic-provider.ts [hunk 1]
```diff
@@ -42,11 +58,24 @@ const defaults: NewRelicProviderOptions = {
...
+export const extras = {
+    preload: preload_newrelic
+}
+
 
 Object.assign(NewrelicProvider, {defaults})
 
 export default NewrelicProvider
 
 if ('undefined' !== typeof (module)) {
```
Comment (rjrodger) [inline src/newrelic-provider.ts:78]: preload will need to be a property of the plugin definition function anyway, so no need to export separately

File: test/fixtures/PluginTestProvider.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+/* Copyright © 2021 Seneca Project Contributors, MIT License. */
+
+
+function PluginTestProvider(this: any, _options: any) {
```
Comment (rjrodger) [inline test/fixtures/PluginTestProvider.ts:4]: remove all uses of the term "provider", as this is not a provider

**Comment by rjrodger** [comment]
I don't see any updates to the test/ordu.js file? this is where I was expecting to see most of your work as per my task document?

**Review by rjrodger — CHANGES_REQUESTED** [review]

File: src/newrelic-provider.ts [hunk 3]
```diff
@@ -42,11 +58,24 @@ const defaults: NewRelicProviderOptions = {
...
+    console.log(plugin);
+    */
+}
+
+export const extras = {
+    preload: preload_newrelic
+}
+
 
 Object.assign(NewrelicProvider, {defaults})
```
[the remark this answers is not in the document]
Comment (vitorclelis96) [reply src/newrelic-provider.ts:74]: Could you please suggest a name?

File: src/newrelic-provider.ts (hunk 1 above)
Comment (vitorclelis96) [reply src/newrelic-provider.ts:78]: OK

File: test/fixtures/PluginTestProvider.ts (hunk 2 above)
Comment (vitorclelis96) [reply test/fixtures/PluginTestProvider.ts:4]: Old

File: test/fixtures/PluginTestProvider.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+    const seneca: any = this
+    const ZONE_BASE = 'provider/other/'
+
+    // NOTE: sys- zone prefix is reserved.
+
+    seneca
+        .message('sys:provider,provider:other,get:info', get_info)
```
[the remark this answers is not in the document]
Comment (vitorclelis96) [reply test/fixtures/PluginTestProvider.ts:11]: OK

File: test/fixtures/PluginTestProvider.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+
+    async function do_stuff(this: any, msg: any) {
+        msg.here = true;
+        return this.prior(this, msg);
+    }
+
+
+    return {
+        exports: {
+            native: () => ({})
```
[the remark this answers is not in the document]
Comment (vitorclelis96) [reply test/fixtures/PluginTestProvider.ts:30]: OK

File: test/newrelic-provider.test.ts [hunk 6]
```diff
@@ -73,6 +72,34 @@ describe('newrelic-provider', () => {
         await (SenecaMsgTest(seneca, NewrelicProviderMessages)())
     })
 
+    test('plugin-test-provider', async () => {
+        const seneca = Seneca({legacy: false})
+            .test()
+            .use('promisify')
+            .use('provider', {
+                provider: {
+                    trello: {
```
[the remark this answers is not in the document]
Comment (vitorclelis96) [reply test/newrelic-provider.test.ts:81]: Removed.

**Pull request closed without merging**